process/rating/ratingsData.go

```diff
@@ -18,70 +21,177 @@ type RatingsData struct {
 			process.ErrStartRatingNotBetweenMinAndMax,
 			settings.General.MinRating,
 			settings.General.StartRating,
 			settings.General.MaxRating)
 	}
 	if settings.General.SignedBlocksThreshold > 1 || settings.General.SignedBlocksThreshold < 0 {
-		return nil, fmt.Errorf("%w signedBlocksThreshold: %v",
+		return fmt.Errorf("%w signedBlocksThreshold: %v",
 			process.ErrSignedBlocksThresholdNotBetweenZeroAndOne,
 			settings.General.SignedBlocksThreshold)
 	}
+	if settings.General.HoursToMaxRatingFromStartRating == 0 {
+		return process.ErrHoursToMaxRatingFromStartRatingZero
+	}
 	if settings.MetaChain.ConsecutiveMissedBlocksPenalty < 1 {
-		return nil, fmt.Errorf("%w: metaChain consecutiveMissedBlocksPenalty: %v",
+		return fmt.Errorf("%w: metaChain consecutiveMissedBlocksPenalty: %v",
 			process.ErrConsecutiveMissedBlocksPenaltyLowerThanOne,
 			settings.MetaChain.ConsecutiveMissedBlocksPenalty)
 	}
 	if settings.ShardChain.ConsecutiveMissedBlocksPenalty < 1 {
-		return nil, fmt.Errorf("%w: shardChain consecutiveMissedBlocksPenalty: %v",
+		return fmt.Errorf("%w: shardChain consecutiveMissedBlocksPenalty: %v",
 			process.ErrConsecutiveMissedBlocksPenaltyLowerThanOne,
 			settings.ShardChain.ConsecutiveMissedBlocksPenalty)
 	}
-	if settings.ShardChain.ProposerDecreaseRatingStep > 0 || settings.ShardChain.ValidatorDecreaseRatingStep > 0 {
-		return nil, fmt.Errorf("%w: shardChain decrease steps - proposer: %v, validator: %v",
-			process.ErrDecreaseRatingsStepPositive,
-			settings.ShardChain.ProposerDecreaseRatingStep,
-			settings.ShardChain.ValidatorDecreaseRatingStep)
+	if settings.ShardChain.ProposerDecreaseFactor > -1 || settings.ShardChain.ValidatorDecreaseFactor > -1 {
+		return fmt.Errorf("%w: shardChain decrease steps - proposer: %v, validator: %v",
+			process.ErrDecreaseRatingsStepMoreThanMinusOne,
+			settings.ShardChain.ProposerDecreaseFactor,
+			settings.ShardChain.ValidatorDecreaseFactor)
 	}
-
-	if settings.MetaChain.ProposerDecreaseRatingStep > 0 || settings.MetaChain.ValidatorDecreaseRatingStep > 0 {
-		return nil, fmt.Errorf("%w: metachain decrease steps - proposer: %v, validator: %v",
-			process.ErrDecreaseRatingsStepPositive,
-			settings.MetaChain.ProposerDecreaseRatingStep,
-			settings.MetaChain.ValidatorDecreaseRatingStep)
+	if settings.MetaChain.ProposerDecreaseFactor > -1 || settings.MetaChain.ValidatorDecreaseFactor > -1 {
+		return fmt.Errorf("%w: metachain decrease steps - proposer: %v, validator: %v",
+			process.ErrDecreaseRatingsStepMoreThanMinusOne,
+			settings.MetaChain.ProposerDecreaseFactor,
+			settings.MetaChain.ValidatorDecreaseFactor)
 	}
+	return nil
+}
 
-	chances := make([]process.SelectionChance, 0)
-	for _, chance := range settings.General.SelectionChances {
-		chances = append(chances, &SelectionChance{
-			MaxThreshold:  chance.MaxThreshold,
-			ChancePercent: chance.ChancePercent,
-		})
+func computeRatingStep(
+	shardSize uint32,
```

Comment:
if there are so many input it would be better to give an argument DTO

Reply:
fixed